ixwebsocket/IXWebSocketTransport.cpp

```diff
@@ -299,16 +300,23 @@ namespace ix
                 }
                 else if (ret <= 0)
                 {
-                    _rxbuf.clear();
                     _socket->close();
+
+                    // if there are received data pending to be processed, then delay the abnormal closure
+                    // to after dispatch (other close code/reason could be read from the buffer)
+                    if (_rxbuf.size() > 0) 
                     {
-                        std::lock_guard<std::mutex> lock(_closeDataMutex);
-                        _closeCode = kAbnormalCloseCode;
-                        _closeReason = kAbnormalCloseMessage;
-                        _closeWireSize = 0;
-                        _closeRemote = true;
+                        _treatAbnormalCloseAfterDispatch = true;
```

Comment:
I don't like that new member variable very much If we need to propagate a state between this function (poll) and dispatch, we should return something in that function (a state, or use a boolean to start). Then this value would be passed in to dispatch as a new parameter.

Reply:
I’ll see that